torchmetrics/image/kid.py

```diff
@@ -0,0 +1,284 @@
+    https://arxiv.org/abs/1801.01401
+
+    [2] GANs Trained by a Two Time-Scale Update Rule Converge to a Local Nash Equilibrium,
+    Martin Heusel, Hubert Ramsauer, Thomas Unterthiner, Bernhard Nessler, Sepp Hochreiter
+    https://arxiv.org/abs/1706.08500
+
+    Raises:
+        ValueError:
+            If ``feature`` is set to an ``int`` (default settings) and ``torch-fidelity`` is not installed
+        ValueError:
+            If ``feature`` is set to an ``int`` not in [64, 192, 768, 2048]
+        ValueError:
+            If ``subsets`` is not an integer larger than 0
+        ValueError:
+            If ``subset_size`` is not an integer larger than 0
+        ValueError:
+            If ``degree`` is not an integer larger than 0
+        ValueError:
+            If ``gamma`` is niether ``None`` or a float larger than 0
+        ValueError:
+            If ``coef`` is not an float larger than 0
+
+    Example:
+        >>> import torch
+        >>> _ = torch.manual_seed(123)
+        >>> from torchmetrics import KID
+        >>> kid = KID(subset_size=50)  # doctest: +SKIP
+        >>> # generate two slightly overlapping image intensity distributions
+        >>> imgs_dist1 = torch.randint(0, 200, (100, 3, 299, 299), dtype=torch.uint8)  # doctest: +SKIP
+        >>> imgs_dist2 = torch.randint(100, 255, (100, 3, 299, 299), dtype=torch.uint8)  # doctest: +SKIP
+        >>> kid.update(imgs_dist1, real=True)  # doctest: +SKIP
+        >>> kid.update(imgs_dist2, real=False)  # doctest: +SKIP
+        >>> kid_mean, kid_std = kid.compute()  # doctest: +SKIP
+        >>> print((kid_mean, kid_std))  # doctest: +SKIP
+        (tensor(0.0338), tensor(0.0025))
+
+    """
+
+    def __init__(
+        self,
+        feature: Union[str, int, torch.nn.Module] = 2048,
+        subsets: int = 100,
+        subset_size: int = 1000,
+        degree: int = 3,
+        gamma: Optional[float] = None,
+        coef: float = 1.0,
+        compute_on_step: bool = False,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Callable = None
+    ):
+        super().__init__(
+            compute_on_step=compute_on_step,
+            dist_sync_on_step=dist_sync_on_step,
+            process_group=process_group,
+            dist_sync_fn=dist_sync_fn,
+        )
+
+        rank_zero_warn(
+            'Metric `KID` will save all extracted features in buffer.'
+            ' For large datasets this may lead to large memory footprint.', UserWarning
+        )
+
+        if isinstance(feature, (str, int)):
+            if not _TORCH_FIDELITY_AVAILABLE:
+                raise ValueError(
+                    'KID metric requires that Torch-fidelity is installed.'
+                    'Either install as `pip install torchmetrics[image-quality]`'
+                    ' or `pip install torch-fidelity`'
+                )
+            valid_int_input = ['logits_unbiased', 64, 192, 768, 2048]
```

Comment:
```suggestion
            valid_int_input = ('logits_unbiased', 64, 192, 768, 2048)
```

use immutables whenever possible